core/src/test/java/org/apache/druid/java/util/common/StringUtilsTest.java

```diff
@@ -148,4 +148,14 @@ public void testReplace()
     Assert.assertEquals("bb", StringUtils.replace("aaaa", "aa", "b"));
     Assert.assertEquals("", StringUtils.replace("aaaa", "aa", ""));
   }
+
+  @Test
+  public void testURLEncodeSpace()
+  {
+    String s1 = StringUtils.urlEncode("aaa bbb");
+    Assert.assertEquals(s1, "aaa%20bbb");
+
+    String s2 = StringUtils.urlEncode("fff+ggg");
+    Assert.assertEquals(s2, "fff%2Bggg");
```

Comment:
also do a decode and verify that the original string is read back ?

Reply:
Added decode checks